lib/src/data/datasource/remote/movie_api_service.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:http/http.dart' show Client;
+
+import '../../../core/util/constants.dart' as constants;
+import '../../../core/util/enums.dart';
+import '../../../domain/entity/genre.dart';
+import '../../../domain/entity/movie.dart';
+import '../../model/genre_model.dart';
+import '../../model/tmdb_item_model.dart';
+
+class MovieApiService {
+  String _getEndpointUrl(MovieEndpoint endpoint) {
+    switch (endpoint) {
+      case MovieEndpoint.topRated:
+        return constants.endpointTopRated;
+      case MovieEndpoint.upcoming:
+        return constants.endpointUpcoming;
+      case MovieEndpoint.nowPlaying:
+        return constants.endpointNowPlaying;
+      case MovieEndpoint.popular:
+        return constants.endpointPopular;
+    }
+  }
+
+  Future<List<Movie>> fetchMovies(
+    int page,
+    MovieEndpoint endpoint,
+  ) async {
+    final Client client = Client();
+    final String url = '${constants.baseMovieUrl}${_getEndpointUrl(endpoint)}';
+
+    try {
+      final response = await client.get(
+        Uri.parse(
+          '$url?page=$page&api_key=${constants.apiKey}',
+        ),
+      );
+      if (response.statusCode == 200) {
+        return TMDBItemModel.fromJson(
+          json.decode(response.body),
+        ).results;
+      } else {
+        throw Exception('Failed to fetch movies');
+      }
+    } catch (e) {
+      rethrow;
+    } finally {
+      client.close();
+    }
+  }
+
+  Future<List<Genre>> fetchGenres() async {
+    final Client client = Client();
+    try {
+      final response = await client.get(
+        Uri.parse(
+          '${constants.baseGenreUrl}?language=en&api_key=${constants.apiKey}',
+        ),
+      );
+      if (response.statusCode == 200) {
+        //
+        final List<Map<String, dynamic>> genreList = List.from(
+          json.decode(response.body)['genres'],
+        );
+        return GenreModel.validGenres = GenreModel.fromJson(
+          genreList,
+        );
+      } else {
+        throw Exception('Failed to get genre data');
+      }
+    } catch (e) {
+      rethrow;
+    } finally {
+      client.close();
+    }
+  }
+
+  Future<List<Movie>> fetchMoviesByGenre(int genre, int page) async {
+    final Client client = Client();
+    final String url =
+        '${constants.discoverMovieUrl}?with_genres=$genre&page=$page';
+
+    try {
+      final response = await client.get(
+        Uri.parse(
+          '$url&api_key=${constants.apiKey}',
+        ),
+      );
+      if (response.statusCode == 200) {
```

Comment:
To replace 200, you can use HttpStatus.OK

Reply:
fixed